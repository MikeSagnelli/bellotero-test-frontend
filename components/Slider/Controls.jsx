import React from 'react';
import PropTypes from 'prop-types';
import { Arrow, RightArrow, SwitchersContainer, SwitchButton } from './styles';

const Counter = ({ previousReview, nextReview }) => (
  <SwitchersContainer>
    <SwitchButton id="prev-button" onClick={previousReview}>
      <Arrow src="/static/images/arrow.svg" alt="next" />
    </SwitchButton>
    <SwitchButton id="next-button" onClick={nextReview}>
      <RightArrow src="/static/images/arrow.svg" alt="next" />
    </SwitchButton>
  </SwitchersContainer>
);

Counter.displayName = 'Counter';

Counter.propTypes = {
  previousReview: PropTypes.func.isRequired,
  nextReview: PropTypes.func.isRequired
};

export default Counter;
