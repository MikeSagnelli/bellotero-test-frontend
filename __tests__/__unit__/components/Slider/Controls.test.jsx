import React from 'react';
import { shallow, mount } from 'enzyme';
import wait from 'waait';
import Controls from '../../../../components/Slider/Controls';

const mockPreviousFunction = jest.fn();
const mockNextFunction = jest.fn();

describe('Slider/Controls', () => {
  it('renders slider', () => {
    const component = shallow(
      <Controls previousReview={mockPreviousFunction} nextReview={mockNextFunction} />
    );
    expect(component).toMatchSnapshot();
  });

  it('calls mockNextFunction', async () => {
    const component = mount(
      <Controls previousReview={mockPreviousFunction} nextReview={mockNextFunction} />
    );

    await wait(0);
    component.update();

    const nextButton = component.find('div#next-button');
    nextButton.simulate('click');
    expect(mockNextFunction).toHaveBeenCalled();
  });

  it('calls mockPreviousFunction', async () => {
    const component = mount(
      <Controls previousReview={mockPreviousFunction} nextReview={mockNextFunction} />
    );

    await wait(0);
    component.update();

    const prevButton = component.find('div#prev-button');
    prevButton.simulate('click');
    expect(mockPreviousFunction).toHaveBeenCalled();
  });
});
