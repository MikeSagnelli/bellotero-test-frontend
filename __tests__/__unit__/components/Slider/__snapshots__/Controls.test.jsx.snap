// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider/Controls renders slider 1`] = `
<styles__SwitchersContainer>
  <styles__SwitchButton
    id="prev-button"
    onClick={[MockFunction]}
  >
    <styles__Arrow
      alt="next"
      src="/static/images/arrow.svg"
    />
  </styles__SwitchButton>
  <styles__SwitchButton
    id="next-button"
    onClick={[MockFunction]}
  >
    <styles__RightArrow
      alt="next"
      src="/static/images/arrow.svg"
    />
  </styles__SwitchButton>
</styles__SwitchersContainer>
`;
